### PR TITLE
chore: Add `*_flat` methods for log events

### PR DESCRIPTION
**Pull request by a-rodin**
Together with #2081 it is supposed to solve the issue of supporting fields with `.` and other special characters.

Not sure if `_flat` suffix is clear enough, or is there a better name for this. I considered `*_top_level` suffix, but discarded it as too long.

It would be interesting to @ktff and @bruceg.

### CODE REVIEW

File: src/event/mod.rs [hunk 1]
```diff
@@ -99,14 +99,26 @@ impl LogEvent {
...
 
+    pub fn get_flat(&self, key: &Atom) -> Option<&Value> {
+        self.fields.get(key)
+    }
+
     pub fn get_mut(&mut self, key: &Atom) -> Option<&mut Value> {
         util::log::get_mut(&mut self.fields, key)
     }
 
+    pub fn get_mut_flat(&mut self, key: &Atom) -> Option<&mut Value> {
```
Comment (bruceg) [inline src/event/mod.rs]: I'm unsure what the right order would be, this or `get_flat_mut`

**Comment by ktff** [comment]
As @bruceg said, it's not clear that this is useful. It at least can't be used in Kubernetes part because only the end of the key is flat, and not all.

**Comment by ktff** [comment]
>  it's not clear that this is useful

Actually forget that, it is useful. Value just needs to be constructed by hand.

**Comment by a-rodin** [comment]
@ktff 

> Value just needs to be constructed by hand.

Note that `Value` implements `From<JsonValue>`, so it might be possible to construct values directly from JSON, if it helps.

**Comment by a-rodin** [comment]
@bruceg 
> I'm unclear where it is meant to be useful.

It is supposed to be used in cases when it is necessary to insert the field at the top level and ignore the special characters like `.` or `[`/`]`. For example, it seems to be userful in `journald` source where the event structure is flat. Currently `journald` source uses `from_records`, which uses `insert`, but because the records in `journald` are always flat, the implementation could have used `insert_flat` instead to improve performance and support names with dots.

**Comment by a-rodin** [comment]
It is going to be used in the `lua` transform, so I'm merging this.

**Pull request merged**